player_linux.go

```diff
@@ -85,6 +86,10 @@ func newPlayer(sampleRate, numChans, bytesPerSample, bufferSizeInBytes int) (*pl
 	return &p, nil
 }
 
+func (p *player) SetUnderrunCallback(f func()) {
+	p.underrun = f
```

Comment:
On player_android.go, writing is done on another goroutine so I thought we need locks at least on Android and might need on other platform for consistency. On second thought, we can use channels for Android so that the callback is called on the groutine of Write. OK, let's go without locks.